Kickstarter-iOS/SharedFunctionsTests.swift

```diff
@@ -0,0 +1,44 @@
+import XCTest
+@testable import Kickstarter_Framework
+
+internal final class SharedFunctionsTests: XCTestCase {
+  func testLogoutAndDismiss() {
+    let mockAppEnvironment = MockAppEnvironment.self
+    let mockPushNotificationDialog = MockPushNotificationDialog.self
+    let mockViewController = MockViewController()
+
+    XCTAssertFalse(mockAppEnvironment.logoutWasCalled)
+    XCTAssertFalse(mockPushNotificationDialog.resetAllContextsWasCalled)
+    XCTAssertFalse(mockViewController.dismissAnimatedWasCalled)
+    mockAppEnvironment.logout()
+    XCTAssertTrue(mockAppEnvironment.logoutWasCalled)
+    mockPushNotificationDialog.resetAllContexts()
```

Comment:
Since we want to test this function

```
func logoutAndDismiss(
  viewController: UIViewController,
  appEnvironment: AppEnvironmentType.Type = AppEnvironment.self,
  pushNotificationDialog: PushNotificationDialogType.Type = PushNotificationDialog.self) { .. }
```

we should simply call it with the mock objects like so

```
logoutAndDismiss(viewController: mockViewController, appEnvironment: mockAppEnvironment, pushNotificationDialog: mockPushNotificationDialog)
```

this will make sure that `AppEnvironment` calls `logout`, `PushNotificationDialog` calls `resetAlContexts()` and `viewController` will call `dismiss(animated: true, completion: nil)`.

So the final test should probably look like this
```
func testLogoutAndDismiss() {
    let mockAppEnvironment = MockAppEnvironment.self
    let mockPushNotificationDialog = MockPushNotificationDialog.self
    let mockViewController = MockViewController()

    XCTAssertFalse(mockAppEnvironment.logoutWasCalled)
    XCTAssertFalse(mockPushNotificationDialog.resetAllContextsWasCalled)
    XCTAssertFalse(mockViewController.dismissAnimatedWasCalled)
    logoutAndDismiss(viewController: mockViewController, appEnvironment: mockAppEnvironment, pushNotificationDialog: mockPushNotificationDialog)
    XCTAssertTrue(mockAppEnvironment.logoutWasCalled)
    XCTAssertTrue(mockPushNotificationDialog.resetAllContextsWasCalled)
    XCTAssertTrue(mockViewController.dismissAnimatedWasCalled)
  }
```